### ISSUE TITLE
Compatibility issue with TypeScript 4.1

### ISSUE THREAD

**Issue by Jack-Works** [naddison36/promievent#5]
```
node_modules/promievent/PromiEvent.ts:28:33 - error TS2345: Argument of type '(resolve: Resolve<T>, reject: Reject) => void' is not assignable to parameter of type '(resolve: (value: T | PromiseLike<T>) => void, reject: (reason?: any) => void) => void'.
  Types of parameters 'resolve' and 'resolve' are incompatible.
    Types of parameters 'value' and 'value' are incompatible.
      Type 'T | PromiseLike<T> | undefined' is not assignable to type 'T | PromiseLike<T>'.
        Type 'undefined' is not assignable to type 'T | PromiseLike<T>'.

28      this.promise = new Promise<T>(executor)
                                      ~~~~~~~~
```

**Comment by naddison36** [comment]
Fixed with version 0.1.4

**Comment by Jack-Works** [comment]
Thank you very much!
